client/src/components/Messager/MessagerContainer.js

```diff
@@ -0,0 +1,169 @@
+import React, { useState, useEffect } from "react";
+import { useAuth } from "hooks/useAuth";
+import { Grid, Hidden, Drawer, makeStyles } from "@material-ui/core";
+
+import Sidebar from "./Sidebar/Sidebar";
+import MessengerMain from "./Main/MessengerMain";
+import Snack from "./Snack";
+
+const useStyles = makeStyles(() => ({
+  root: {
+    maxHeight: "100vh",
+    maxWidth: "100vw",
+    overflow: "hidden",
+  },
+
+  drawerPaper: {
+    width: "350px",
+  },
+}));
+
+export default function MessagerContainer() {
+  const [isSideOpen, setIsSideOpen] = useState(false);
+  const [currentConvo, setCurrentConvo] = useState(null);
+  const [allConvos, setAllConvos] = useState(null);
+  const [snack, setSnack] = useState(null);
+
+  const classes = useStyles();
+  const auth = useAuth();
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const response = await fetch("/api/convos", {
+          method: "get",
+        });
+        if (response.ok) {
+          let conversations = await response.json();
+          conversations = conversations.map((c) => assignRandomImage(c));
+          setAllConvos(conversations);
+          setCurrentConvo(conversations[0]);
+          setSnack({
+            message: `Welcome back ${auth.user.username}!`,
+            severity: "success",
+          });
+        } else {
+          setSnack({
+            message: "Something went wrong on our end",
+            severity: "error",
+          });
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    getData();
+  }, []);
+
+  // Creates conversations with users after search
+  const createConversation = async (recipient) => {
+    try {
+      const response = await fetch("/api/convos", {
+        method: "post",
+        body: JSON.stringify({ recipient: recipient._id }),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      if (response.ok) {
+        const data = await response.json();
+        let conversation = {
+          ...data.conversation,
+          users: [auth.user, recipient],
```

Comment:
It mostly is but the users are only saved as refs rather than whole document so Im not populating before sending back because I already have the user's data